Document current known foreign key fixes possibly required post upgrading

VVelox opened the pull request:
Pulled the required fixes from various issues in regards to possible foreign key issues that can arise post updating.

VVelox committed:
add current known foriegn key fixes possibly required post upgrading

MrPetovan commented on doc/Update.md:
```diff
@@ -77,3 +77,32 @@ RENAME TABLE <table_name>_new TO <table_name>;
+
+### Resolving Possible Database Issues Post Upgrading
+
+#### Foriegn Keys
```
```suggestion
#### Foreign Keys
```

MrPetovan commented on doc/Update.md:
```diff
@@ -77,3 +77,32 @@ RENAME TABLE <table_name>_new TO <table_name>;
+
+#### Foriegn Keys
+
+Some of the updates include the use of foreign keys now that will bump
+into issues with previous versions, which would sometimes shove bad
+data into tables, preventing, causing errors such as below.
```
For Markdown document, we have a convention where we write a sentence a line.
```suggestion
Some of the updates include the use of foreign keys now that will bump into issues with previous versions, which would sometimes shove bad data into tables, preventing, causing errors such as below.
```

MrPetovan commented on doc/Update.md:
````diff
@@ -77,3 +77,32 @@ RENAME TABLE <table_name>_new TO <table_name>;
+delete from diaspora-interaction where uri-id not in (select id from item-uri);
+```
+
+This all has been compiled as of currently from issue #9746, #9753,
+and #9878.
````
```suggestion
This all has been compiled as of currently from issue #9746, #9753, and #9878.
```

MrPetovan commented on doc/Update.md:
````diff
@@ -77,3 +77,32 @@ RENAME TABLE <table_name>_new TO <table_name>;
+All current known fixes for possible items that can go wrong are as below.
+
+```SQL
+delete from item where `owner-id` not in (select id from contact);
+delete from item where `contact-id` not in (select id from contact);
+delete from notify where `uri-id` not in (select id from `item-uri`);
+delete from photo where `contact-id` not in (select id from contact);
+delete from thread where iid not in (select id from item);
+delete from item where `author-id` not in (select id from contact);
+delete from diaspora-interaction where uri-id not in (select id from item-uri);
````
```suggestion
DELETE FROM `item` WHERE `owner-id` NOT IN (SELECT `id` FROM `contact`);
DELETE FROM `item` WHERE `contact-id` NOT IN (SELECT `id` FROM `contact`);
DELETE FROM `notify` WHERE `uri-id` NOT IN (SELECT `id` FROM `item-uri`);
DELETE FROM `photo` WHERE `contact-id` NOT IN (SELECT `id` FROM `contact`);
DELETE FROM `thread` WHERE `iid` NOT IN (SELECT `id` FROM `item`);
DELETE FROM `item` WHERE `author-id` NOT IN (SELECT `id` FROM `contact`);
DELETE FROM `diaspora-interaction` WHERE `uri-id` NOT IN (SELECT `id` FROM `item-uri`);
```

MrPetovan committed:
Improve formatting in doc/Update.md

- Fix typo
- Enforce Markdown convention of a sentence per line
- Improve SQL formatting regarding language keywords and escaped identifiers

MrPetovan reviewed (approved):
Thank you for your work!
